src/mocks/timeseriesDataPointsList.ts

```diff
@@ -2,8 +2,87 @@ import {
   DatapointsGetAggregateDatapoint,
   DatapointsGetDoubleDatapoint,
 } from '@cognite/sdk';
+import { MockChartDataConfig } from '.';
 
-export function randomData(
+export const defaultConf: MockChartDataConfig = {
+  id: 0,
+  min: 0,
+  max: 150,
+  continousDeviation: 3,
+  peakDeviation: 30,
+};
+
+export function getRandomdata(
+  id: number,
+  start: number,
+  end: number,
+  n: number,
+  granularity?: number,
+  config?: any
+) {
+  let conf = config ? getConig(config, id) : undefined;
+  conf = conf ? conf : defaultConf;
+  const arr = randomData(start, end, n, conf, granularity);
+  return makePeakPoints(arr, conf);
+}
+
+function getConig(config: any[], id: number) {
+  return config.find(conf => conf.id === id);
+}
+
+// tslint:disable-next-line:cognitive-complexity
+function makePeakPoints(
+  resultArray: DatapointsGetAggregateDatapoint,
+  config: MockChartDataConfig
+) {
+  let sortedPoints = [...resultArray.datapoints].sort((dp1, dp2) => {
```

Comment:
I can't understand why this complex logic is here. Could you please simplify it? Just increase `negativePeakPoints` and `positivePeakPoints` number of points with `peakDeviation` value in a random position on the timeline.

Reply:
I used this function to improve the smoothness of the chart. So this finds all “distinct auto-generated peak points” and improves them with “max peak deviation” given with configuration. The data generated was closer to actual data with this function.

Chart with this function
![Screenshot from 2020-01-23 11-35-38](https://user-images.githubusercontent.com/17356270/72966468-0f329600-3de5-11ea-9aaf-823ff27e997d.png)

Chart with random peak points
![Screenshot from 2020-01-23 11-35-36](https://user-images.githubusercontent.com/17356270/72966479-19549480-3de5-11ea-9c10-64f17d53d909.png)